src/enginearrays.jl

```diff
@@ -21,8 +21,8 @@ function engine(p::AbstractEngineArray;bcdims=nothing)
     end
 end
 Base.parent(p::EngineArray) = p.parent
-DiskArrays.readblock!(a::EngineArray,xout,r::OrdinalRange...) = DiskArrays.readblock!(a.parent,xout,r...)
-DiskArrays.writeblock!(a::EngineArray,xout,r::OrdinalRange...) = DiskArrays.writeblock!(a.parent,xout,r...)
+DiskArrays.readblock!(a::EngineArray,xout,r::OrdinalRange...) = a.parent isa AbstractDiskArray ? DiskArrays.readblock!(a.parent,xout,r...) : xout .= a.parent[r...]
```

Comment:
should this be `isdisk(a.parent)` instead?

Reply:
That sounds about right.